sklearn/model_selection/tests/test_search.py

```diff
@@ -1775,3 +1777,15 @@ def get_n_splits(self, *args, **kw):
                              'inconsistent results. Expected \\d+ '
                              'splits, got \\d+'):
         ridge.fit(X[:train_size], y[:train_size])
+
+
+def test_n_features_in():
+    # make sure grid search delegates n_features_in to the best estimator
+    n_features = 4
+    X, y = make_classification(n_features=n_features)
+    gbdt = HistGradientBoostingClassifier()
```

Comment:
I thought we used `DummyClassifier` for these cases. I see what you did here :P

Reply:
Unfortunately we can't here, because the dummies don't validate the input (so `n_features_in_` isn't set for them).

Reply:
hmm, I would argue that the `DummyClassifier` should be the simplest classifier satisfying our API. And since we're adding requirements to our public API, the `DummyClassifier` should also implement them.

Reply:
The thing is, the dummies really don't use X at all (by design)

This is hilariously wrong but this code works just fine:

```py
DummyClassifier().fit(['any', 'thing', 'that', 'is the same size as y'],
                      [0, 1, 2, 3])
                .predict([dict(), (1,)])  # output = [0, 2]
```

> DummyClassifier should be the simplest classifier satisfying our API.

I agree with that, but on the other side it seems artificial to enforce a specific structure for X when we end up not actually using it.